lib/network/README.md

```diff
@@ -0,0 +1,33 @@
+# Network library
+
+This is the internal contributor documentation.
+
+## Goals
+
+The main goal of the network library is to provide utility functions for IPv4 and IPv6 conversions.
+It should have the following properties.
+
+- Easy: The functions should have obvious semantics.
+- Safe: Throw early and helpful errors when mistakes are detected.
+- Lazy: Only compute values when necessary.
+
+## Tests
+
+Tests are declared in `tests.sh`.
+
+## Internal Representation
+
+`IPv4Address`
+
+An attribute set with these values:
+
+- `cidr`: A CIDR.
+- `address`: An IP address.
+- `prefixLength`: A prefix length.
+- `subnetMask`: A subnet mask.
```

Comment:
It doesn't sound like this is internal, in which case it shouldn't be here but rather in the user documentation (which you can put at the top of the `default.nix`, like [here](https://github.com/NixOS/nixpkgs/blob/master/lib/fileset/default.nix#L1)) :)

Reply:
Done! Thanks for the feedback; helps me understand how the libraries are structured and organized.